joomla/joomla-cms
# [6.2] Allowing menu items to have multiple 'rel' values

### Summary of Changes

[According to MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link#attr-rel), the `rel` attribute on `<link>` and `<a>` elements can have multiple space-separated values.

This PR makes this a little easier by turning the rel field into a fancy select list and allows multiple values, which get imploded into a string before getting passed back to the layout (so there shouldn't be B/C problems).

### Testing Instructions

For this PR to be a success a few things must be true:

1. Existing `rel` fields must be converted over without loss of data.
2. You must be able to add multiple link type values and have them show up on the link in the front end.

#### To test:

1. **BEFORE applying the patch**, make a menu item of type URL and go to the "Link Type" tab. Select a link type from the "link rel attribute" dropdown.
2. Go to the front end and make sure the link is displaying correctly, with the selected attribute getting rendered in the HTML. (It should look something like `<a href="url here" rel="selected-attribute">`.)
3. Apply the patch.
4. **Success criteria 1:** Go to the front end and make sure the link is still displaying the same way as before.
5. Go back to the link type settings and select multiple attributes. Save.
6.  **Success criteria 2:** Go to the front end and make sure the selected attributes display in the html, separated by spaces. (It should look something like `<a href="url here" rel="selected-attribute another-attr attribute-three">`.)

### Actual result BEFORE applying this Pull Request

Could not select multiple rel attributes.


### Expected result AFTER applying this Pull Request

Existing links are unchanged; new capability of selecting multiple rel attributes works well.

### Link to documentations
Please select:


I don't know yet whether documentation is needed.
